build/bcd-urls.js

```diff
@@ -0,0 +1,116 @@
+    // https://developer.mozilla.org/docs/Web/API/MediaTrackSettings/width
+    // So to get the appropriate slug, in Yari, we have to assume a locale.
+    let slug = u.pathname;
+    if (slug.startsWith("/docs/")) {
+      // Important! For now, to make this a slug we can understand we
+      // have to have a locale and we pick `en-US` as the default.
+      return `/en-US${slug}`;
+    }
+    throw new Error(`not implemented! ${slug}`);
+  }
+
+  doc.body
+    .filter((section) => section.type === "browser_compatibility")
+    .forEach((section) => {
+      Object.entries(section.value.data).forEach(([key, block]) => {
+        // First block from the BCD data does not have its name as the root key
+        // so mdn_url is accessible at the root. If the block has a key for
+        // `__compat` it is not the first block, and the information is nested
+        // under `__compat`.
+        if (block.__compat) {
+          block = block.__compat;
+        }
+        if (block.mdn_url) {
+          const url = getPathFromAbsoluteURL(block.mdn_url);
+          block.mdn_url = url;
+          if (checkLinks && url.startsWith("/")) {
+            // Now we need to scrutinize if that's a url we can fully
+            // recognize. (But only if it's a relative link)
+            const urlLC = url.toLowerCase();
+            const found = Document.findByURL(urlLC, {
+              metadata: true,
+            });
+            const query = section.value.query;
+            if (!found) {
+              // That means trouble!
+              // But can it be salvaged with a redirect?
+              const resolved = Redirect.resolve(urlLC);
+              // Remember that `Redirect.resolve()` will return the input
+              // if it couldn't be resolved to a *different* url.
+              if (resolved && resolved !== urlLC) {
+                // Just because it's a redirect doesn't mean it ends up
+                // on a page we have.
+                // For example, there might be a redirect but where it
+                // goes to is not in this.allTitles.
+                // This can happen if it's a "fundamental redirect" for example.
+                const finalDocument = Document.findByURL(resolved, {
+                  metadata: true,
+                });
+                const suggestion = finalDocument ? finalDocument.url : null;
+                addBadBCDLink(query, key, url, suggestion);
+                block.mdn_url = suggestion;
+              } else {
+                addBadBCDLink(query, key, url);
+                // If the url is bad, and can't be salvaged with a redirect,
+                // we need to pass that information on so that it can be
+                // leveraged in the UI that displays the BCD table.
+                // That way, it knows to not attempt to make a link out of it.
+                block.bad_url = true;
+              }
+            }
+          }
+        }
+      });
+    });
+}
```

Comment:
Not a fan of how heavily nested this is, the lines above look like lisp.

Here's an attempt at simplification:
```javascript
const sections = doc.body
  .filter((section) => section.type === "browser_compatibility")
  .flatMap((section) => Object.entries(section.value.data));

for (let [key, block] of sections) {
  // First block from the BCD data does not have its name as the root key
  // so mdn_url is accessible at the root. If the block has a key for
  // `__compat` it is not the first block, and the information is nested
  // under `__compat`.
  if (block.__compat) {
    block = block.__compat;
  }
  if (!block.mdn_url) {
    continue;
  }
  const url = getPathFromAbsoluteURL(block.mdn_url);
  block.mdn_url = url;
  if (!(checkLinks && url.startsWith("/"))) {
    continue;
  }
  const urlLC = url.toLowerCase();
  const found = Document.findByURL(urlLC, {
    metadata: true,
  });
  const query = section.value.query;
  if (found) {
    continue;
  }
  const resolved = Redirect.resolve(urlLC);
  if (resolved && resolved !== urlLC) {
    // Just because it's a redirect doesn't mean it ends up
    // on a page we have.
    // For example, there might be a redirect but where it
    // goes to is not in this.allTitles.
    // This can happen if it's a "fundamental redirect" for example.
    const finalDocument = Document.findByURL(resolved, {
      metadata: true,
    });
    const suggestion = finalDocument ? finalDocument.url : null;
    addBadBCDLink(query, key, url, suggestion);
    block.mdn_url = suggestion;
  } else {
    addBadBCDLink(query, key, url);
    // If the url is bad, and can't be salvaged with a redirect,
    // we need to pass that information on so that it can be
    // leveraged in the UI that displays the BCD table.
    // That way, it knows to not attempt to make a link out of it.
    block.bad_url = true;
  }
}
```

Reply:
See, that's why healthy code review is so healthy. 
I could have written that (neater) code but I was so busy trying to write down the business logic desires into code that I ran out of energy to re-paint the style of the code. 
I'll see about cleaning it up.